diff --git a/node_modules/http-errors/index.js b/node_modules/http-errors/index.js
index 10ca4ad..1d5ae4b 100644
--- a/node_modules/http-errors/index.js
+++ b/node_modules/http-errors/index.js
@@ -55,7 +55,7 @@ function createError () {
     var arg = arguments[i]
     if (arg instanceof Error) {
       err = arg
-      status = err.status || err.statusCode || status
+      status = err.status || err.statusCode || err.code || status
       continue
     }
     switch (typeof arg) {
@@ -70,6 +70,7 @@ function createError () {
         break
       case 'object':
         props = arg
+        status = props.status || props.statusCode || props.code || status
         break
     }
   }
